pintc/src/main.rs

```diff
@@ -47,7 +47,18 @@ fn main() -> anyhow::Result<()> {
         }
     };
 
-    match handler.scope(|handler| compile_contract(handler, &flattened)) {
+    // Optimise
+    let contract = if args.skip_optimise {
+        flattened
+    } else {
+        let optimised = flattened.optimise();
```

Comment:
If you're going to separate `optimize` from `compile()` above, then you also need to call `optimise()` in `pink-pkg`.. search for where `compile(..)` is used.

Reply:
In fact, I think `optimize` should be part of `compile()`.. just make it easier. In tests, you can separate the steps which is what we do (`type_check`, then `flatten`, then `optimize`)

Reply:
Do we not want the user to be able to skip the optimise step when compiling? Should it only be optional for tests?

Reply:
Good question.  I'd say optimisation shouldn't be optional for regular compiling, although it would be useful for development -- for us.  Perhaps a hidden flag to disable optimisations?  Are hidden options possible with `clap`?

Reply:
`hide` is in the [documentation](https://docs.rs/clap/latest/clap/struct.Arg.html#method.hide) for `arguments`. I haven't found anything on hiding `options`.

I spent some time working on hiding `skip-optimise` but I couldn't get it to work. I can hide any `help` text for the option, but not the option itself.

ex. 
```rust
#[arg(long = "skip-optimise", help="skip optimisation phase", hide=true)]
    pub skip_optimise: bool,
```
just hides "skip optimisation phase".

```
Usage: pintc [OPTIONS] <FILEPATH>

Arguments:
  <FILEPATH>  

Options:
  -o, --output <OUTPUT>  
      --print-parsed     
      --print-flat       
      --print-asm        
      --print-optimised  
  -h, --help             Print help
```

Reply:
Isn't this what we want?  I don't see `--skip-optimise' in the output there.

BTW, I think for docs we're meant to use the 'ize' suffix.  I've accidentally let a couple of 'ise's through by habit.

Reply:
Well that's embarrassing. Yes, it is working and I kept reading `--print-optimised` as `--skip-optimised`. Will add it now.

> BTW, I think for docs we're meant to use the 'ize' suffix. I've accidentally let a couple of 'ise's through by habit.

haha! I figured we'd keep in the `ise` in honor of you being the first to say we should make it a new phase. I'll also update it to `ize`